lib/ledger_sync/resource_attribute.rb

```diff
@@ -72,5 +74,11 @@ def will_change?(val)
       assert_valid(value: val)
       value != type.cast(value: val)
     end
+
+    def ensure_references_many_value_presence(args)
+      if references_many? && args.fetch(:value) == nil
+        args[:value] = []
+      end
+    end
```

Comment:
Basically, when an array is not received in the answer (in my exemple it's amazon business api), `references_many` is `nil`. Its value is `nil` too. I ensure here that if we meet this specific case, we set an empty array instead `[]`